Fault Event and Issue Report updates

Lytrix opened the pull request:
- Updated/Added new fault events up to commit 8bab297 of events.md
- Updated IssueReport Faultevents with hex and total insulin delivered values
- Updated PodInfoTest for new values

Lytrix committed:
- Updated/Added new fault events up to commit 8bab297 of events.md
- Updated IssueReport Faultevents with hex and total insulin delivered values
- Updated PodInfoTest for new values

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -96,27 +97,28 @@ public struct PodInfoFaultEvent : PodInfo, Equatable {
-            "* podMessageCounter: \(podMessageCounter)",
-            "* unknownPageCode: \(unknownPageCode.hexadecimalString)",
+            "* insulinNotDelivered: (%.2f, \(insulinNotDelivered)) U",
+            "* podMessageCounter: 0x\(podMessageCounter)",
```
The 0x prefix implies, hex, but this is rendering decimal

Lytrix replied:
kept decimal to render, only used it for unknownValue

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -16,13 +16,13 @@ public struct PodInfoFaultEvent : PodInfo, Equatable {
     public let currentStatus: FaultEventCode
     public let faultEventTimeSinceActivation: TimeInterval?
-    public let reservoirLevel: Double?
+    public let reservoirLevel: String
```
The internal rep of this should stay `Double?`.  Conversion to string form should be done at display time.

Lytrix replied:
reverted back to Double? and formatted at display time.

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -96,27 +97,28 @@ public struct PodInfoFaultEvent : PodInfo, Equatable {
-            "* insulinNotDelivered: \(insulinNotDelivered) U",
-            "* podMessageCounter: \(podMessageCounter)",
-            "* unknownPageCode: \(unknownPageCode.hexadecimalString)",
+            "* insulinNotDelivered: (%.2f, \(insulinNotDelivered)) U",
```
I don't think this is doing what you are intending. It will render like `* totalInsulinDelivered: (%.2f, 0.0) U`

Lytrix replied:
made an extension for this.

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -61,16 +62,16 @@ public struct PodInfoFaultEvent : PodInfo, Equatable {
         self.timeActive = TimeInterval(minutes: Double(encodedData[13...14].toBigEndian(UInt16.self)))
         
-        self.previousStatus = FaultEventCode(rawValue: encodedData[15])
+        self.unacknowledgedAlerts = Data(encodedData[15])
```
If this is a single value, then it should be an integer type.

Lytrix replied:
fixed

Lytrix committed:
- Only prepend 0x hex for still unknown values, if decimal render without any.
- Format reservoirLevel as String type only at display time
- Single hex: type as Integer
- Force 2 decimals on all insulin values
- Updated Podinfo tests

Lytrix commented:
Thanks for the fast reviewing @ps2 ! I've updated the parts that were wrong. 
I only haven't been able to test if reservoirLevel renders default as Optional("50+") or as "50+".

ps2 commented on OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift:
```diff
@@ -63,14 +64,14 @@ public struct PodInfoFaultEvent : PodInfo, Equatable {
         self.timeActive = TimeInterval(minutes: Double(encodedData[13...14].toBigEndian(UInt16.self)))
         
-        self.previousStatus = FaultEventCode(rawValue: encodedData[15])
+        self.unacknowledgedAlerts = encodedData[15]
```
Does this field map to AlertType?

ps2 replied:
@itsmojo This is the TT field.  Should it map to these types?https://github.com/ps2/rileylink_ios/blob/f0191d2805d782a920f16b816d20c69f9c2e8b07/OmniKit/MessageTransport/MessageBlocks/ConfigureAlertsCommand.swift#L34-L42

itsmojo replied:
Yes, as long as Loop continues to the same conventions for the alerts that the PDM uses.

ps2 replied:
In the [$19 Configure Alerts](https://github.com/openaps/openomni/wiki/Command-19-Configure-Alerts) command, the alarm type is not a bitmask, but a single value, which we decode to the `AlertType` shown above.  The [wiki for this TT field](https://github.com/openaps/openomni/wiki/Command-02-Pod-Information-Response#type-2) indicates it's a bitmask, so are the `AlertType` values translated into bit positions and used as a bitmask, or is this field the same as the `aaaaaaaa` bits in the $1d response?  The latter is decoded into PodAlarm (https://github.com/ps2/rileylink_ios/blob/bea1f163bc98f092ef9e6ee9e42480fb14377635/OmniKit/Model/PodAlarm.swift#L11-L19)

ps2 replied:
Sounds like we need to do a bigger refactor here; PodAlarm and AlertTypes should be based off of the same values; we can take that up in a different PR.

ps2 commented on OmniKitTests/PodInfoTests.swift:
```diff
@@ -225,15 +225,15 @@ class PodInfoTests: XCTestCase {
+            XCTAssertEqual(0, decoded.insulinNotDelivered)
             XCTAssertEqual(4, decoded.podMessageCounter)
-            XCTAssertEqual("07eb", decoded.unknownPageCode.hexadecimalString)
+            XCTAssertEqual("50.15", decoded.totalInsulinDelivered.twoDecimals)
```
Probably better to test actual value here, than string representation.  The .twoDecimals method is a private extension for formatting the debugDescription, not for external use.

itsmojo commented:
For the type2 fault info, the insulinNotDelivered and totalInsulinDelivered fields don't need and should not have any bit masking as they are both returned as 16-bit word values. This is unlike the $1D response where these same values are squeezed into 11-bit and 13-bit fields for minimal message size (heck, they even dropped the message length byte for the $1D response to make it smaller). And in any case, the unneeded insulinNotDelivered bit masking for the type2 fault info was incorrect as this is actually an 11-bit value.
